lib/database_cleaner/active_record/transaction.rb

```diff
@@ -3,20 +3,19 @@
 module DatabaseCleaner
   module ActiveRecord
     class Transaction < Base
-      def start
-        # Hack to make sure that the connection is properly set up before cleaning
-        connection_class.connection.transaction {}
+      Error = Class.new(StandardError)
 
-        connection_class.connection.begin_transaction joinable: false
-      end
-
-
-      def clean
-        connection_class.connection_pool.connections.each do |connection|
-          next unless connection.open_transactions > 0
-          connection.rollback_transaction
-        end
+      case ::ActiveRecord::VERSION::MAJOR
+      when 5
+        require 'database_cleaner/active_record/rails5'
+        include DatabaseCleaner::ActiveRecord::Rails5
+      when 6
+        require 'database_cleaner/active_record/rails6'
+        include DatabaseCleaner::ActiveRecord::Rails6
+      else
+        raise Error, "Major Rails version #{::Rails::VERSION::MAJOR} is unsupported by database_cleaner"
```

Comment:
@ryanong If I understand this correctly, users using Rails 7 will get an Error. Is there a way to improve this code so that it will work with Rails 7, 8, n...?